test/utils/utils.go

```diff
@@ -66,7 +66,7 @@ func StartEmbeddedEtcd(ctx context.Context, etcdDir string, logger *logrus.Entry
 	cfg.InitialCluster = cfg.InitialClusterFromName(cfg.Name)
 	cfg.Logger = "zap"
 	cfg.AutoCompactionMode = "periodic"
-	cfg.AutoCompactionRetention = "5m"
+	cfg.AutoCompactionRetention = "0"
```

Comment:
Any particular reason for disabling auto-compaction? Gardener etcds usually have auto-compaction enabled, which is why keeping it enabled here mimics the behavior of production etcds to the closest extent possible.

Reply:
There was an flaky error coming due to this. I dedicated this PR only for fixing the build. I think, further improvement of the test case should be done in another PR. It would be good for future as well.